core/modules/parser/QSMySqlListener.cc

```diff
@@ -1526,26 +1571,22 @@ class BinaryComparasionPredicateAdapter :
         auto compPredicate = make_shared<query::CompPredicate>();
         compPredicate->left = _left;
 
-        // We need to remove the coupling between the query classes and the parser classes, in this case where
-        // the query classes use the integer token types instead of some other system. For now this if/else
-        // block allows us to go from the token string to the SqlSQL2Tokens type defined by the antlr2/3
-        // grammar and used by the query objects.
         if ("=" == _comparison) {
-            compPredicate->op = SqlSQL2Tokens::EQUALS_OP;
+            compPredicate->op = query::CompPredicate::EQUALS_OP;
```

Comment:
Interesting, I find the single line if statements to be less readable.

Reply:
This is probably another "philosophical" discussion. When I have a code like this (repeated structure, and minor variations in statements) I often find it more readable when I make it look like a table. This hast two benefits: 1) one can "scroll up/down" a "column" to see differences in rows, and 2) it's easy to cross check a condition in the left column ('if (...)`) versus a statement in the right one. Both help in the "eye-code-analysis"

Reply:
I would say that if you want to make your code look like a table, instead of relying on a waiver of the code formatting guidelines, you should rather create an actual table and make the code into a table lookup.

Reply:
Yes, indeed. This code fragment is exactly the case for using a `std::map<std::string, query::CompPredicate::WHATEVER_ENUM_DEF>`.

Reply:
Probably via some form of:
```
static const std::map<std::string, query::CompPredicate::WHATEVER_ENUM_DEF> = ...;

query::CompPredicate::WHATEVER_ENUM_DEF str2predicate(string str) {
    auto itr = str2predMap.find(str);
    if (nullptr == itr) throw std::range_error("no such predicate:" + str);
    return its.second;
}
```